[2025-06-13T17:11:14.286+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T16:07:32.404313+00:00 [queued]>
[2025-06-13T17:11:14.322+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T16:07:32.404313+00:00 [queued]>
[2025-06-13T17:11:14.322+0200] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-06-13T17:11:14.323+0200] {taskinstance.py:1280} INFO - Starting attempt 4 of 4
[2025-06-13T17:11:14.323+0200] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-06-13T17:11:14.358+0200] {taskinstance.py:1300} INFO - Executing <Task(BashOperator): pyspark_train_classifier_model> on 2025-06-11 16:07:32.404313+00:00
[2025-06-13T17:11:14.371+0200] {standard_task_runner.py:55} INFO - Started process 32313 to run task
[2025-06-13T17:11:14.380+0200] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'agile_data_science_batch_prediction_model_training', 'pyspark_train_classifier_model', 'manual__2025-06-11T16:07:32.404313+00:00', '--job-id', '51', '--raw', '--subdir', 'DAGS_FOLDER/setup.py', '--cfg-path', '/tmp/tmpmpfmn81u']
[2025-06-13T17:11:14.381+0200] {standard_task_runner.py:83} INFO - Job 51: Subtask pyspark_train_classifier_model
[2025-06-13T17:11:14.540+0200] {task_command.py:388} INFO - Running <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T16:07:32.404313+00:00 [running]> on host l089.lab.dit.upm.es
[2025-06-13T17:11:14.716+0200] {taskinstance.py:1507} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=agile_data_science_batch_prediction_model_training
AIRFLOW_CTX_TASK_ID=pyspark_train_classifier_model
AIRFLOW_CTX_EXECUTION_DATE=2025-06-11T16:07:32.404313+00:00
AIRFLOW_CTX_TRY_NUMBER=4
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-06-11T16:07:32.404313+00:00
[2025-06-13T17:11:14.718+0200] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2025-06-13T17:11:14.718+0200] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\nspark-submit --master local[4]   /home/monica.fernandez/practica_creativa//resources/train_spark_mllib_model.py   /home/monica.fernandez/practica_creativa/']
[2025-06-13T17:11:14.730+0200] {subprocess.py:86} INFO - Output:
[2025-06-13T17:11:18.679+0200] {subprocess.py:93} INFO - 25/06/13 17:11:18 WARN Utils: Your hostname, l089 resolves to a loopback address: 127.0.1.1; using 138.4.31.89 instead (on interface enp3s0)
[2025-06-13T17:11:18.682+0200] {subprocess.py:93} INFO - 25/06/13 17:11:18 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-06-13T17:11:23.991+0200] {subprocess.py:93} INFO - 25/06/13 17:11:23 INFO SparkContext: Running Spark version 3.5.3
[2025-06-13T17:11:23.992+0200] {subprocess.py:93} INFO - 25/06/13 17:11:23 INFO SparkContext: OS info Linux, 6.1.0-32-amd64, amd64
[2025-06-13T17:11:23.993+0200] {subprocess.py:93} INFO - 25/06/13 17:11:23 INFO SparkContext: Java version 17.0.14
[2025-06-13T17:11:24.078+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-06-13T17:11:24.268+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO ResourceUtils: ==============================================================
[2025-06-13T17:11:24.269+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-06-13T17:11:24.271+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO ResourceUtils: ==============================================================
[2025-06-13T17:11:24.272+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO SparkContext: Submitted application: train_spark_mllib_model.py
[2025-06-13T17:11:24.342+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-06-13T17:11:24.363+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO ResourceProfile: Limiting resource is cpu
[2025-06-13T17:11:24.371+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-06-13T17:11:24.507+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO SecurityManager: Changing view acls to: monica.fernandez
[2025-06-13T17:11:24.509+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO SecurityManager: Changing modify acls to: monica.fernandez
[2025-06-13T17:11:24.513+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO SecurityManager: Changing view acls groups to:
[2025-06-13T17:11:24.514+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO SecurityManager: Changing modify acls groups to:
[2025-06-13T17:11:24.515+0200] {subprocess.py:93} INFO - 25/06/13 17:11:24 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: monica.fernandez; groups with view permissions: EMPTY; users with modify permissions: monica.fernandez; groups with modify permissions: EMPTY
[2025-06-13T17:11:25.159+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO Utils: Successfully started service 'sparkDriver' on port 41561.
[2025-06-13T17:11:25.217+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO SparkEnv: Registering MapOutputTracker
[2025-06-13T17:11:25.284+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO SparkEnv: Registering BlockManagerMaster
[2025-06-13T17:11:25.320+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-06-13T17:11:25.322+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-06-13T17:11:25.328+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-06-13T17:11:25.364+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-31fe567f-3207-4aa4-abd7-2f0442fea358
[2025-06-13T17:11:25.382+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-06-13T17:11:25.402+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-06-13T17:11:25.596+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-06-13T17:11:25.795+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-06-13T17:11:25.809+0200] {subprocess.py:93} INFO - 25/06/13 17:11:25 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2025-06-13T17:11:26.054+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO Executor: Starting executor ID driver on host 138.4.31.89
[2025-06-13T17:11:26.055+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO Executor: OS info Linux, 6.1.0-32-amd64, amd64
[2025-06-13T17:11:26.055+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO Executor: Java version 17.0.14
[2025-06-13T17:11:26.076+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-06-13T17:11:26.077+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@403f70d6 for default.
[2025-06-13T17:11:26.109+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 38929.
[2025-06-13T17:11:26.109+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO NettyBlockTransferService: Server created on 138.4.31.89:38929
[2025-06-13T17:11:26.112+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-06-13T17:11:26.118+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 138.4.31.89, 38929, None)
[2025-06-13T17:11:26.130+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO BlockManagerMasterEndpoint: Registering block manager 138.4.31.89:38929 with 434.4 MiB RAM, BlockManagerId(driver, 138.4.31.89, 38929, None)
[2025-06-13T17:11:26.136+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 138.4.31.89, 38929, None)
[2025-06-13T17:11:26.139+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 138.4.31.89, 38929, None)
[2025-06-13T17:11:26.863+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-06-13T17:11:26.872+0200] {subprocess.py:93} INFO - 25/06/13 17:11:26 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmpdni6yra3/spark-warehouse'.
[2025-06-13T17:11:28.882+0200] {subprocess.py:93} INFO - MLflow Run ID: d7f1c8bef41b4d96a01fb92b9d039798, Experiment ID: 0
[2025-06-13T17:11:28.882+0200] {subprocess.py:93} INFO - MLflow Tracking URI: file:///tmp/airflowtmpdni6yra3/mlruns
[2025-06-13T17:11:29.882+0200] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-06-13T17:11:29.882+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 216, in <module>
[2025-06-13T17:11:29.882+0200] {subprocess.py:93} INFO -     main(sys.argv[1])
[2025-06-13T17:11:29.883+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 65, in main
[2025-06-13T17:11:29.884+0200] {subprocess.py:93} INFO -     features = spark.read.json(input_path, schema=schema)
[2025-06-13T17:11:29.884+0200] {subprocess.py:93} INFO -                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-06-13T17:11:29.884+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 425, in json
[2025-06-13T17:11:29.884+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-06-13T17:11:29.884+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/.sdkman/candidates/spark/current/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-06-13T17:11:29.912+0200] {subprocess.py:93} INFO - pyspark.errors.exceptions.captured.AnalysisException: [PATH_NOT_FOUND] Path does not exist: file:/data/simple_flight_delay_features.jsonl.bz2.
[2025-06-13T17:11:30.335+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO SparkContext: Invoking stop() from shutdown hook
[2025-06-13T17:11:30.336+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-06-13T17:11:30.355+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO SparkUI: Stopped Spark web UI at http://138.4.31.89:4041
[2025-06-13T17:11:30.373+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-06-13T17:11:30.389+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO MemoryStore: MemoryStore cleared
[2025-06-13T17:11:30.389+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO BlockManager: BlockManager stopped
[2025-06-13T17:11:30.399+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-06-13T17:11:30.403+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-06-13T17:11:30.423+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO SparkContext: Successfully stopped SparkContext
[2025-06-13T17:11:30.424+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO ShutdownHookManager: Shutdown hook called
[2025-06-13T17:11:30.425+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO ShutdownHookManager: Deleting directory /tmp/spark-6d7acfbf-4903-4e41-ae21-c7dfaeab9c70
[2025-06-13T17:11:30.429+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO ShutdownHookManager: Deleting directory /tmp/spark-e10b8d1d-e4ed-4736-8481-b264704713ed/pyspark-e26b5e4a-1fa7-447f-bb4b-20dc0f25b5bc
[2025-06-13T17:11:30.434+0200] {subprocess.py:93} INFO - 25/06/13 17:11:30 INFO ShutdownHookManager: Deleting directory /tmp/spark-e10b8d1d-e4ed-4736-8481-b264704713ed
[2025-06-13T17:11:30.490+0200] {subprocess.py:97} INFO - Command exited with return code 1
[2025-06-13T17:11:30.531+0200] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/airflow/operators/bash.py", line 196, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-06-13T17:11:30.537+0200] {taskinstance.py:1318} INFO - Marking task as FAILED. dag_id=agile_data_science_batch_prediction_model_training, task_id=pyspark_train_classifier_model, execution_date=20250611T160732, start_date=20250613T151114, end_date=20250613T151130
[2025-06-13T17:11:30.590+0200] {standard_task_runner.py:100} ERROR - Failed to execute job 51 for task pyspark_train_classifier_model (Bash command failed. The command returned a non-zero exit code 1.; 32313)
[2025-06-13T17:11:30.603+0200] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-06-13T17:11:30.637+0200] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
